hphp/runtime/vm/bytecode.cpp

```diff
@@ -6321,14 +6348,26 @@ void recordCodeCoverage(PC /*pc*/) {
       unit == SystemLib::s_hhas_unit) {
     return;
   }
+
+  const StringData* filepath = unit->filepath();
+  assert(filepath->isStatic());
+  const char* filename = filepath->data();
+
+  if (unit != s_prev_unit) {
+    s_prev_unit = unit;
+    if (RID().getCoverageWithUnused() || RID().getCoverageWithDeadCode()) {
+      recordExecutableLines(unit, filename);
+    }
+    if (RID().getCoverageWithDeadCode()) {
+      recordNotDeadCode(unit, filename);
+    }
+  }
+
```

Comment:
This could be removed.  The filter for initializing a new m_hits entry would handle this.
Also this will recompute the DeadCode, and ExecutableLines every time we switch back to a unit.